pkg/agent/statusupdate.go

```diff
@@ -68,18 +86,41 @@ func (su *periodicStatusUpdate) Execute() error {
 	return nil
 }
 
-// StartPeriodicStatusUpdate - starts a job that runs the periodic status updates
-func StartPeriodicStatusUpdate() {
+// StartAgentStatusUpdate - starts 2 separate jobs that runs the periodic status updates and immediate status updates
+func StartAgentStatusUpdate() {
+	startPeriodicStatusUpdate()
+	startImmediateStatusUpdate()
+}
+
+// startPeriodicStatusUpdate - start periodic status updates based on report activity frequency config
+func startPeriodicStatusUpdate() {
 	interval := agent.cfg.GetReportActivityFrequency()
-	statusUpdate = &periodicStatusUpdate{}
-	_, err := jobs.RegisterDetachedIntervalJob(statusUpdate, interval)
+	periodicStatusUpdate = &agentStatusUpdate{
+		typeOfStatusUpdate: periodic,
+	}
+	_, err := jobs.RegisterDetachedIntervalJob(periodicStatusUpdate, interval)
```

Comment:
This one may not need to be detached anymore, since it is only for normal execution status updates

Reply:
updated